syntax = "proto3";

package nebio;

message Event
{
    string event_id             = 1;
    string event_type           = 2;
    uint64 time                 = 3;
    string page                 = 4;
    string referer              = 5;
    string session_id           = 6;
    string user_id              = 7;
    string device_id            = 8;
    string plat                 = 9;
    string explorer             = 10;
    string client_ip            = 11;
    uint32 app_id               = 12;

    // add by analyse
    E_EVENT_OPERATE event_oper  = 13;
    string tag                  = 14;
    string tourist_id           = 15;
    string next_page            = 16;
    uint32 vv                   = 17;
    uint32 exit_vv              = 18;
    uint32 bounce_vv            = 19;
    uint64 length               = 20;       // 时长（单位：S）
    int32  user_type            = 21;       // 用户类型

    enum E_EVENT_OPERATE
    {
        EVENT_ADD               = 0;        // 新增事件
        EVENT_DEL               = 1;        // 删除事件
    }
}

message Result
{
    uint32 app_id               = 1;
    string date                 = 2;
    string channel              = 3;
    string tag                  = 4;
    
    string key1                 = 5;
    string key2                 = 6;
    string key3                 = 7;
    string key4                 = 8;
    uint32 key5                 = 9;
    uint32 key6                 = 10;

    // value
    uint32 pv                   = 11;
    uint32 uv                   = 12;
    uint32 vv                   = 13;
    uint32 iv                   = 14;
    uint64 length               = 15;
    uint32 exit_vv              = 16;
    uint32 bounce_vv            = 17;
}

